druid/src/contexts.rs

```diff
@@ -378,14 +387,24 @@ impl<'a> EventCtx<'a> {
 impl<'a> LifeCycleCtx<'a> {
     #[deprecated(since = "0.5.0", note = "use request_paint instead")]
     pub fn invalidate(&mut self) {
-        self.base_state.needs_inval = true;
+        self.request_paint();
     }
 
-    /// Request a [`paint`] pass.
+    /// Request a [`paint`] pass. This is equivalent to calling [`request_paint_rect`] for the
+    /// entire layout rectangle.
```

Comment:
I think we want the paint rect rather than the layout rect here. But the basic idea of using existing geometry is sound.

Reply:
Got it.